### PR TITLE
fix OTRS-1000029719 and OTRS-1000040550

### PR DESCRIPTION
**Pull request by hajgato**
Fixes https://github.com/hpcugent/slurm-torque-wrappers/issues/10 and https://github.com/hpcugent/slurm-torque-wrappers/issues/9

### CODE REVIEW

**Comment by boegel** [comment]
@hajgato I'd love to review this, but I don't know Perl well enough to do a proper job at it...

Maybe @stdweird can take a quick look?

File: files/qsub.pl [hunk 1]
```diff
@@ -551,9 +548,22 @@ sub parse_script
     };
 
 
-    # Look for PBS directives for -o, -e, -N, -j
+    # Look for PBS directives for -o, -e, -N, -j if command they not given in the command line
```
Comment (stdweird) [inline files/qsub.pl]: i don't understand the comment

File: files/qsub.pl [hunk 2]
```diff
@@ -551,9 +548,22 @@ sub parse_script
...
     # If they are not set, add the commandline args from defaults
     # keys are slurm option names
+    my %orig_args = map { $_ => 1 } @orig_args;
```
Comment (stdweird) [inline files/qsub.pl]: let's see if i get this right
```perl
my %orig_args = map { s/^-//; $_ => 1 } grep {m/^-.$/} @orig_args;  # only map the one-letter options, and remove the leading -
my @check_pbsopt = qw(j o N);
push (@check_opts, 'e') if $orig_args{j};
@check_pbsopt = grep {!$orig_args{$_}} @check_pbsopt;
```

File: files/qsub.pl [hunk 3]
```diff
@@ -573,19 +583,26 @@ sub parse_script
             }
         }
     };
+
     # if -j PBS directive is in the script,
     # do not use default error path for slurm
     my @check_eo = qw(e o);
-    if ($set{'j'}) {
+    if ($set{'j'} || $orig_args{'-j'}) {
```
Comment (stdweird) [inline files/qsub.pl]: this should not be needed here.
the defaults is already handled with line below, and `make_command` should only set defaults, not the actual `-e` command.

File: files/qsub.pl [hunk 4]
```diff
@@ -660,7 +677,7 @@ sub main
 
     # stdin is not relevant for interactive jobs
     # but should also add the defaults
-    ($stdin, $command) = parse_script($stdin, $command, $defaults);
+    ($stdin, $command) = parse_script($stdin, $command, $defaults, @orig_args);
```
Comment (stdweird) [inline files/qsub.pl]: need to pass the `@orig_args` as arrayref, so pass `\@orig_args`

File: files/qsub.pl [hunk 5]
```diff
@@ -521,7 +518,7 @@ sub run_submitfilter
 
 sub parse_script
 {
-    my ($txt, $command, $defaults) = @_;
+    my ($txt, $command, $defaults, @orig_args) = @_;
```
Comment (stdweird) [inline files/qsub.pl]: use `$orig_args` as arayref

File: files/qsub.pl (hunk 2 above)
Comment (stdweird) [reply files/qsub.pl]: ok, and with arrayref `$orig_args`, you can't use `%orig_args` as name (otherwise `$orig_args` is not what you think it is)

**Comment by hajgato** [comment]
@stdweird Your modifications resulted:
1) 254 failed tests
2) multiple "Use of uninitialized value $_ in pattern match (m//) at files/qsub.pl line 554."

File: files/qsub.pl (hunk 2 above)
Comment (hajgato) [reply files/qsub.pl]: @stdweird `@orig_args` can be empty.

**Comment by hajgato** [comment]
@stdweird But it seems to be my fault.

**Comment by hajgato** [comment]
@stdweird Everything looks fine now.

**Pull request merged**